### ISSUE TITLE
Relax 1051 for simple condition

### ISSUE THREAD

**Issue by jnyrup** [JosefPihrt/Roslynator#169]
The following triggers 1051 about parenthesizing the condition:
```csharp
var result = boolVariable ? 1 : 2;
```

giving:
```csharp
var result = (boolVariable) ? 1 : 2;
```

Could this be relaxed to not trigger when the condition is just a boolean expression?

**Comment by JosefPihrt** [comment]
Hi, I decided to keep it as it is now.

1) It would be a breaking change.
2) "No exceptions" is a better solution than making a list of exceptions (Someone could argue that simple invocation `Foo()` could be also excluded).


This issue can be reopen if there will be any mechanism how to pass an options to an analyzer.

**Comment by JosefPihrt** [comment]
Fixed with ffb3b97c4e4e8ddb018be006ce240236c0fd82e1
